src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/customize_panel/customize_panel_editor.tsx

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+
+import {
+  EuiFormRow,
+  EuiFieldText,
+  EuiSwitch,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiForm,
+  EuiTextArea,
+  EuiFlyoutFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSuperDatePicker,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { TimeRange } from '@kbn/es-query';
+import { hasTimeRange, TimeRangeInput } from './customize_panel_action';
+import { doesInheritTimeRange } from './does_inherit_time_range';
+import { IEmbeddable, Embeddable, EmbeddableOutput, CommonlyUsedRange } from '../../../..';
+
+type PanelSettings = {
+  title?: string;
+  hidePanelTitles?: boolean;
+  description?: string;
+  timeRange?: TimeRange;
+};
+
+interface CustomizePanelProps {
+  embeddable: IEmbeddable;
+  dateFormat?: string;
+  commonlyUsedRanges?: CommonlyUsedRange[];
+  onClose: () => void;
+}
+
+const VISUALIZE_EMBEDDABLE_TYPE = 'visualization';
+
+type VisualizeEmbeddable = IEmbeddable<{ id: string }, EmbeddableOutput & { visTypeName: string }>;
+
+function isVisualizeEmbeddable(
+  embeddable: IEmbeddable | VisualizeEmbeddable
+): embeddable is VisualizeEmbeddable {
+  return embeddable.type === VISUALIZE_EMBEDDABLE_TYPE;
+}
+
+function isTimeRangeCompatible(embeddable: IEmbeddable) {
+  const isInputControl =
```

Comment:
This is good for now - especially since it's copied from `ui_actions_enhanced`. This is one of those places where we might eventually use an interface to avoid leaking visualize embeddable details into the embeddable plugin.